### PR TITLE
Refactor more speech functions

### PR DESCRIPTION
**Pull request by feerrenrut**
### Link to issue number:
None

### Summary of the issue:
Returning sequences from these functions allows us greater flexibility for affecting all speech that is caused by a call to speakObject.

### Description of how this pull request fixes the issue:
Split several functions into a speakX and getXspeech

This PR maintains backwards compatibility

### Testing performed:
Ran NVDA from source.

### Known issues with pull request:
- Todo: rename getSpeechForSpelling to getSpellingSpeech

### Change log entry:

Section: New features, Changes, Bug fixes

### CODE REVIEW

**Comment by zstanecic** [comment]
Hi Reef,

Will this change affect drivers, which are already ported to py3?

Note that codefactory closed porting of their voalizer.

Will this affect RHVoice or any other tts?

We should start thinking about reputation among users.

The more backward compatible changes we have, the less is the possibility that they will want to upgrade to the new NVDA version.

Now we have this situation with py3.

Isn’t it better to leave this for NVDA 2020?

This is just my thinking and questions, along with concerns.

Sorry, but if we start to bring more and more backward compatible changes now, it will break us, and we will be in the situation that will lead to fragmentation, and people will prefer staying on NVDA 2019.2.1, because it’s older and add-ons work flawlessly.

**Comment by michaelDCurran** [comment]
@zstanecic : These changes should not affect synthDrivers at all.

**Comment by Brian1Gaff** [comment]
There should, in my view be a neutral view on this, at least for some time, 
since there will be more annoyance from people who have updated whose 
existing maybe home scripted solutions suddenly stop working than from the 
fragmentation of users. After all, I and no doubt others still have some old 
hardware running XP as useful extras, which use a very old version of nvda, 
and that has not really worried the progression much, as long as people keep 
their old add ons.
 Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal E-mail to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
Newsgroup monitored: alt.comp.blind-users
----- Original Message -----

**Comment by feerrenrut** [comment]
@zstanecic and @Brian1Gaff This PR does not break backwards compatibility. The description has been updated to make that clear.

File: source/speech/__init__.py [hunk 1]
```diff
@@ -379,34 +412,44 @@ def speakObjectProperties(  # noqa: C901
...
+		reason: str = controlTypes.REASON_QUERY,
+		_prefixSpeechCommand: Optional[SpeechCommand] = None,
+		priority: Optional[Spri] = None
+):
+	sequence = getObjectSpeech(
+		obj,
+		reason,
+		_prefixSpeechCommand,
+	)
+	speak(sequence, priority=priority)
```
Comment (leonardder) [inline source/speech/__init__.py]: May be do a similar if check as in the other functions?

**Review by LeonarddeR — APPROVED** [review]
This looks good to me, however note that I only had to had time to look at the code briefly, I haven't tested this. I therefore recommend a look from @michaelDCurran before merging this.

**Review by michaelDCurran — CHANGES_REQUESTED** [review]
* Would it be possible to make a new  type annotation for 'reason'? All through this code you have reason: str. It would be great if this could be its own type. 
* with all the speakSelection functions:  it would be good if they could take speech sequences, rather than just strings.  Then speakSelectionChange can pass in a sequence from getTextInfoSpeech rather than just TextInfo.text. It also means that some of the code in speakSelectionChange can be replaced with a call to getSpellingSpeech. However, these changes would mean we can't have the text included with the message as part of one translatable string. Though I don't believe this is a bad thing.

**Comment by michaelDCurran** [comment]
Is it just me, or have the changes to ```speech/__init__.py``` been lost in this pr?

**Comment by feerrenrut** [comment]
> Is it just me, or have the changes to speech/__init__.py been lost in this pr?

I think it's just you 😉. It is hidden from the diff initially because it is "large". You can jump to it, or other files by pressing 't' and typing '__init__'. A direct link to the file in the diff: https://github.com/nvaccess/nvda/pull/10593/files#diff-46301b9ce4a78abc208b3358db03e972

**Review by michaelDCurran — APPROVED** [review]

**Comment by feerrenrut** [comment]
I'm going to rebase and merge this PR (to preserve the history of the refactor). I'll push after the rebase so that the commits in this PR / branch match.

**Pull request merged**